azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java

```diff
@@ -88,38 +138,77 @@ private FrontendIPConfiguration createFrontendIPConfig(String name) {
 
     // Withers (fluent)
 
-    @Override
-    public LoadBalancerImpl withExistingPublicIpAddress(PublicIpAddress publicIpAddress) {
+    private LoadBalancerImpl withExistingPublicIpAddress(PublicIpAddress publicIpAddress) {
         return this.withExistingPublicIpAddress(publicIpAddress.id());
     }
 
-    @Override
-    public LoadBalancerImpl withExistingPublicIpAddress(String resourceId) {
+    private LoadBalancerImpl withExistingPublicIpAddress(String resourceId) {
         FrontendIPConfiguration frontendIpConfig = createFrontendIPConfig(null);
         SubResource pip = new SubResource();
         pip.withId(resourceId);
         frontendIpConfig.withPublicIPAddress(pip);
         return this;
     }
 
-    // Getters
+    @Override
+    public LoadBalancerImpl withNewPublicIpAddress() {
+        // Autogenerated DNS leaf label for the PIP
+        String dnsLeafLabel = this.name().toLowerCase().replace("\\s", "");
+        return withNewPublicIpAddress(dnsLeafLabel);
+    }
 
-    private void ensureCreationPrerequisites() {
+    @Override
+    public LoadBalancerImpl withNewPublicIpAddress(String dnsLeafLabel) {
+        Creatable<PublicIpAddress> creatablePIP = myManager().publicIpAddresses().define(dnsLeafLabel)
+                .withRegion(this.regionName())
+                .withExistingResourceGroup(this.resourceGroupName());
+        return withNewPublicIpAddress(creatablePIP);
```

Comment:
Just confirming..

Suppose LB is defined using `withNewResourceGroup(..)` and with no pip, this means LB has dependency on RG - the our DAG logic creates RG first then LB.

Suppose LB also has dependency on a Pip, that means Pip needs to be created before LB. Now in the DAG there are two leaf nodes `Creatable` RG and `Creatable` Pip, since Pip is not marked as having dependency on `Creatable` RG there is no guarantee that RG will be created first. 

See this code handing similar scenario.. 

https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java#L581